cfme/infrastructure/config_management.py

```diff
@@ -463,96 +684,102 @@ def step(self):
 
     def resetter(self):
         if self.obj.appliance.version >= '5.8' and self.obj.type == 'Ansible Tower':
-            accordion.tree('Providers', 'All Ansible Tower Providers')
-        else:
-            accordion.tree('Providers', 'All Configuration Manager Providers')
-        tb.select('Grid View')
-
-    def am_i_here(self):
-        if self.obj.appliance.version >= '5.8' and self.obj.type == 'Ansible Tower':
-            page = 'All Ansible Tower Providers'
+            self.view.sidebar.providers.tree.click_path('All Ansible Tower Providers')
         else:
-            page = 'All Configuration Manager Providers'
-        return match_page(summary=page)
+            self.view.sidebar.providers.tree.click_path('All Configuration Manager Providers')
 
 
 @navigator.register(ConfigManager, 'Add')
 class MgrAdd(CFMENavigateStep):
+    VIEW = ConfigAddView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        cfg_btn('Add a new Provider')
+        self.prerequisite_view.toolbar.configuration.item_select('Add a new Provider')
 
 
 @navigator.register(ConfigManager, 'Edit')
 class MgrEdit(CFMENavigateStep):
+    VIEW = ConfigEditView
     prerequisite = NavigateToSibling('All')
 
     def step(self):
-        sel.check(Quadicon(self.obj.quad_name, None).checkbox())
-        cfg_btn('Edit Selected item')
+        provider_name = '{} Configuration Manager'.format(self.obj.name)
```

Comment:
**REQUIRED**: Perhaps we should make this **name** be a property on the provider so yo ucan call it via self.obj.ui_name or something. I originally had this as optional, but then it became too important.

Reply:
Will do.